server/http/controller.go

```diff
@@ -737,32 +754,30 @@ func (c *controller) buildDataStores(ctx context.Context, info openapi.ResolveRe
 			}}, ds...)
 		}
 
-		if context.Selector != "" {
-			selector, err := selectors.New(context.Selector)
-			if err != nil {
-				return []expression.DataStore{}, err
-			}
-
-			spans := selector.Filter(*run.Trace)
-			ds = append([]expression.DataStore{expression.MetaAttributesDataStore{
-				SelectedSpans: spans,
-			}}, ds...)
-		}
-	}
-
-	if context.EnvironmentId != "" {
-		environment, err := c.testDB.GetEnvironment(ctx, context.EnvironmentId)
-
+		selector, err := selectors.New(context.Selector)
 		if err != nil {
-			return []expression.DataStore{}, err
+			return [][]expression.DataStore{}, err
 		}
 
-		ds = append([]expression.DataStore{expression.EnvironmentDataStore{
-			Values: environment.Values,
+		spans := selector.Filter(*run.Trace)
+		ds = append([]expression.DataStore{expression.MetaAttributesDataStore{
+			SelectedSpans: spans,
 		}}, ds...)
+
+		if context.SpanId == "" {
+			dsList := make([][]expression.DataStore, len(spans))
+			for i, span := range spans {
+
+				dsList[i] = append([]expression.DataStore{expression.AttributeDataStore{
+					Span: span,
+				}}, ds...)
+			}
+
+			return dsList, nil
+		}
 	}
 
-	return ds, nil
+	return [][]expression.DataStore{ds}, nil
```

Comment:
Now we are returning an array of strings, this is to cover the scenario where there is no span id in the context and we can return the different resolved values from the selected spans.